bot/src/main.rs

```diff
@@ -779,7 +817,7 @@ fn classify_producers(
         }
     }
     let cluster_average_skip_rate = 100 - total_blocks * 100 / total_slots;
-    for (validator_identity, (blocks, slots)) in blocks_and_slots {
+    for (validator_identity, (blocks, slots)) in blocks_and_slots.clone() {
```

Comment:
Try this instead to avoid the clone:
```suggestion
    for (validator_identity, (blocks, slots)) in &blocks_and_slots {
```
which might then require adding some dereferences (`*`) in the block, but that's ok